### PR TITLE
Support pointers of wrapping structs

### PR DESCRIPTION
**Pull request by idaunis**

### CODE REVIEW

**Comment by jackc** [comment]
If I'm reading this right it will assign to the first field in a struct that is a primitive type (`kindTypes`). Which means that nearly any struct could be assigned to as long as it had a primitive field. So I could do something like the following without error?

```go
	var foo struct{
		A, B, C int32
	}

	conn.QueryRow(ctx, "select 42").Scan(&foo)
```

`A` would have `42` assigned to it.

Maybe I'm reading it wrong and that's not the case, but if so that seems way too brittle and error prone.

**Comment by idaunis** [comment]
I narrowed to convert only single and anonymous field struct. Thank you.

File: uuid.go [hunk 1]
```diff
@@ -85,6 +85,9 @@ func (src *UUID) AssignTo(dst interface{}) error {
 		case *string:
 			*v = encodeUUID(src.Bytes)
 			return nil
+		case *Text:
+			v.Set(encodeUUID(src.Bytes))
+			return nil
```
Comment (jackc) [inline uuid.go:90]: Is this part necessary?

**Comment by jackc** [comment]
The main implementation is more reasonable now, but TBH it still makes me a little queasy. There's already a lot of magic and reflection in pgtype to support scanning to types that are slightly "misshaped" and this adds another. Oh well, I guess if there's no other way we can do it... Is there really no other way? If not let me know on the `*Text` branch of UUID and I can merge it.

File: uuid.go (hunk 1 above)
Comment (idaunis) [reply uuid.go]: Not really, I remove it. There is not need to use it in this way from a `pgtype.Text`

**Comment by idaunis** [comment]
I can not think of any other way, I cleaned up and refactored a bit. Let me know what you think.

**Comment by georgysavva** [comment]
Hello guys! Is this PR related to the original problem described in https://github.com/jackc/pgx/issues/1000?
From what I can see in the code and tests and doesn't fix that issue.

**Comment by idaunis** [comment]
Yes, it does fixes the original problem for us with wrapping UUID or wrapping custom strings the implement Scan. What doesn't fix is the `pgtype.Text` case for scanning a UUID, but I don't think it applies.

**Comment by georgysavva** [comment]
BTW guys, why don't you want to make this logic of unwrapping the struct recursive? @jackc @idaunis

**Pull request merged**